board: Add NXP VMU RT1170 board support package

bperseghetti opened the pull request:
Adds support for the NXP VMU RT1170 board. This Vehicle Management Unit based on the i.MX RT1176 brings a fantastic combination of sensors and IO all on one board for development of various systems. It is also the featured board for CogniPilot's Cerebri - VMU autopilot software based on Zephyr.

bperseghetti committed:
board: Add NXP VMU RT1170 board support package

Adds support for the NXP VMU RT1170 board. This Vehicle
Management Unit based on the i.MX RT1176 brings a fantastic
combination of sensors and IO all on one board for development
of various systems. It is also the featured board for
CogniPilot's Cerebri - VMU autopilot software based on Zephyr.

Co-authored-by: Peter van der Perk <peter.vanderperk@nxp.com>
Signed-off-by: Benjamin Perseghetti <bperseghetti@rudislabs.com>

bperseghetti commented:
@dleach02 @danieldegrasse @mmahadevan108 here is our PR for the VMU RT1170 board.

bperseghetti committed:
soc: nxp_rt11xx: add unique PWM names.

Enable PWM to use unique device names.

Signed-off-by: Benjamin Perseghetti <bperseghetti@rudislabs.com>

decsny commented on dts/arm/nxp/nxp_rt11xx.dtsi:
```diff
@@ -527,7 +527,7 @@
 			interrupts = <129 0>;
 
-			flexpwm1_pwm0: pwm0 {
+			flexpwm1_pwm0: flexpwm1_pwm0 {
```
can you explain the reason for this change

bperseghetti replied:
Yes, this is to match what was also changed in the [rt10xx.dtsi](https://github.com/zephyrproject-rtos/zephyr/commit/176d51555cdbe7da107d757cea9ed8d80e220512). The problem becomes that you can have multiple pwmx devices on same channel meaning that you cannot individually address them as expected or needed.

bperseghetti committed:
soc: nxp_rt11xx: add CANFD compatible.

Enable CANFD for rt11xx by including nxp,flexcan-fd
compatibility for all CANFD capable CAN with associated
properties.

Signed-off-by: Benjamin Perseghetti <bperseghetti@rudislabs.com>

igalloway commented:
VMURT1176 is the combination of a SOM module VMUMRT1176 and a Baseboard supporting among other things like 100BaseT1 Automotive Ethernet.  (other PIxhawk standard and customg base boards will also work)

bperseghetti commented:
@mmahadevan108 @dleach02 @danieldegrasse @DerekSnell @manuargue  when I run this locally on my machine it passes. Yet it seems to fail in CI:
https://github.com/zephyrproject-rtos/zephyr/actions/runs/5790244946/job/15692992818?pr=61192#step:13:1595
https://github.com/zephyrproject-rtos/zephyr/actions/runs/5790244946/job/15692992818?pr=61192#step:13:1594

git log to show it's same as what just ran:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/10233412/d3541c5b-12e6-4c30-a175-403ed75b7846)

Running it locally:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/10233412/06a080d7-c6c6-41c1-a93d-05911e7f8671)

manuargue commented:
> @mmahadevan108 @dleach02 @danieldegrasse @DerekSnell @manuargue when I run this locally on my machine it passes. Yet it seems to fail in CI: https://github.com/zephyrproject-rtos/zephyr/actions/runs/5790244946/job/15692992818?pr=61192#step:13:1595 https://github.com/zephyrproject-rtos/zephyr/actions/runs/5790244946/job/15692992818?pr=61192#step:13:1594

I've retriggered the failed job, let's see if it was an sporadic failure.

bperseghetti committed:
tests: net/socket/socketpair: exclude vmu and 1160

This has failed in repeated CI runs. Filed an issue
for having it checked. Excluding it until it's
fixed.

Signed-off-by: Benjamin Perseghetti <bperseghetti@rudislabs.com>

mmahadevan108 commented on boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts:
```diff
@@ -75,7 +75,7 @@ zephyr_mipi_dsi: &mipi_dsi {
 	status = "okay";
 	bus-speed = <125000>;
-
+	bus-speed-data = <1000000>;
```
Was `can-fd` tested?

bperseghetti replied:
We did test it, It's been a while though. I can flash and test again.

bperseghetti replied:
It comes from needing to set FD compatibility in the nxp_rt11xx.dtsi

bperseghetti replied:
And once that gets added you have to include that in the higher level mimxrt1170_evk_cm7.dts as well.

decsny replied:
> We did test it, It's been a while though. I can flash and test again. That value was selected as the default as it can be used as well with a smaller DLC to send CAN2.0 frames over FD.

If you made a lot of changes to the branch since testing it then please test again

gramsay0 commented:
Does anyone know why this board use `CONFIG_TICKLESS_KERNEL=n`?

I only ask because this board failed an assertion I was adding to cortex_m_systick.c https://github.com/zephyrproject-rtos/zephyr/pull/66601#issuecomment-1858985146, and not many boards use that configuration.


The board overflows a calculation due to the high system clock frequency.
Increasing `CONFIG_SYS_CLOCK_TICKS_PER_SEC` _**may**_ be an alternative to running in ticked mode
